### PR TITLE
feat(builder): use normalized config

### PR DESCRIPTION
**Pull request by Asuka109**
# PR Details

## Description

Refactor the way config used in the plugin. Try to use normalized config instead of raw config from users as possible.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by jkzing — COMMENTED** [review]
+1

File: packages/builder/builder-webpack-provider/src/plugins/startUrl.ts [hunk 1]
```diff
@@ -14,19 +15,21 @@ export function PluginStartUrl(): BuilderPlugin {
...
-          const urls = Array.isArray(startUrl) ? startUrl : [startUrl];
-          for (const url of urls) {
-            await open(url);
-          }
-        } else if (startUrl === true) {
-          const { default: open } = await import('../../compiled/open');
-          const protocol = config.dev?.https ? 'https' : 'http';
-          await open(`${protocol}://localhost:${port}`);
+        const config = api.getNormalizedConfig();
+        const { default: open } = await import('../../compiled/open');
```
Comment (chenjiahan) [inline packages/builder/builder-webpack-provider/src/plugins/startUrl.ts:19]: 这里性能劣化了，没配 startUrl 的时候不应该引用 open 模块

**Comment by chenjiahan** [comment]
代码冲突需要 resolve 一下哈

File: packages/builder/builder-webpack-provider/src/plugins/startUrl.ts (hunk 1 above)
Comment (Asuka109) [reply packages/builder/builder-webpack-provider/src/plugins/startUrl.ts:19]: 好滴我改下

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**